editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -5949,13 +5949,14 @@ bool CanvasItemEditorViewport::can_drop_data(const Point2 &p_point, const Varian
 						continue;
 					}
 					Ref<PackedScene> scn = res;
+					Ref<Texture2D> tex = res;
 					if (scn.is_valid()) {
 						Node *instantiated_scene = scn->instantiate(PackedScene::GEN_EDIT_STATE_INSTANCE);
 						if (!instantiated_scene) {
 							continue;
 						}
 						memdelete(instantiated_scene);
-					} else {
+					} else if (!tex.is_valid()) {
```

Comment:
When does this actually happen? My understanding was that we already check for extensions for texture above, and it's already checked for being a valid RES. What valid RES with a texture extension wouldn't be a texture?

Reply:
The problem was if the `res` was not a PackedScene, it would `continue` and never set `can_instantiate` to true.

The code can be simplified I guess.